### PR TITLE
Controlled mode support and fix hasValue method for complex values

### PR DESCRIPTION
**Pull request by delphin92**

### CODE REVIEW

File: src/ReactSelectMaterialUi.tsx [hunk 1]
```diff
@@ -95,6 +95,15 @@ class ReactSelectMaterialUi extends React.Component<ReactSelectMaterialUiProps,
 		);
 	}
 
+	public componentWillReceiveProps(props: ReactSelectMaterialUiProps) {
```
Comment (iulian-radu-at) [inline src/ReactSelectMaterialUi.tsx]: As componentWillReceiveProps is deprecated, I cannot accept code changes using it.
[https://reactjs.org/docs/react-component.html#updating](url)

File: .gitignore [hunk 2]
```diff
@@ -1,4 +1,3 @@
 .history/
 .vscode/
 node_modules/
-lib/
```
Comment (iulian-radu-at) [inline .gitignore]: Why did you removed lib from .gitignore? We do not want to store generated files.
Sorry, I cannot accept a PR adding generated files to git.

Comment (delphin92) [reply .gitignore]: I'm new ib GitHub... Can I do PR from cpecific branch? Unfortunately, I need to have the generated files in master to use my repo as a dependency.

Please, tell me how I can solve this problem correctly.

**Comment by iulian-radu-at** [comment]
I fixed the controlled mode in version 2.0.0. The fix for complex versions will be made in a further version.

**Pull request closed without merging**